controllers/useraccount/useraccount_controller.go

```diff
@@ -475,6 +484,13 @@ func (r *Reconciler) deleteUser(ctx context.Context, userAcc *toolchainv1alpha1.
 	logger := log.FromContext(ctx)
 	logger.Info("deleting the User resources")
 
+	//get the UID before deleting the user
+	// TO DO: this is a temporary solution to remove resources created for users logging in from OIDC which don't have owner references after 4.15
+	uid := userList[0].UID
+	logger.Info(fmt.Sprintf("checking for user settings resources with UID [%s] to be deleted", uid))
+	if deleted, err := r.deleteUserResources(ctx, string(uid)); err != nil || deleted {
+		return deleted, err
```

Comment:
I don't think we can return here. The idea behind deleting the User resource first and stop the reconcile (by return an empty result here: https://github.com/codeready-toolchain/member-operator/pull/568/files#diff-bda4e43a8c3e78834044bcb623b8e1a57e184bec0d940e8dd0f534446796ab69R139 ) is the following:
- we delete the user and stop
- since the controller watches Users it will get triggered again when the User is deleted
- we delete the identities
- since the controller watches Identities it will get triggered again when the Identity is deleted

But it won't work with user setting resources because we do not watch them. So I think we should proceed with deleting the User resource here and return only in case of an error.

Reply:
makes sense, thanks for explaining this. I've made the change here -[1bf005e](https://github.com/codeready-toolchain/member-operator/pull/568/commits/1bf005eeb31a8b6968271c667f5b5777c5f4c969) - to delete all the user resources in the same reconcile before deleting User, and only returns if there is an error.